subsys/bluetooth/mesh/dfu_cli.c

```diff
@@ -305,6 +305,21 @@ static void tx_end(int err, void *cb_data)
 	blob_cli_broadcast_tx_complete(&cli->blob);
 }
 
+static int tx(struct bt_mesh_model *mod, struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf,
+	      const struct bt_mesh_send_cb *cb, struct bt_mesh_dfu_cli *cli)
+{
+	int err;
+
+	err = bt_mesh_model_send(mod, ctx, buf, cb, cli);
+	if (err) {
+		LOG_ERR("Send err: %d", err);
+		tx_end(err, cli);
```

Comment:
Should it be like this?
```suggestion
		if (cb) {
			tx_end(err, cli);
		}
```

Reply:
Or perhaps even more generally, `if (cb) { cb->end(err, cli); }` ?

Reply:
True